a981d3738014339cd0fae1e7f5aa9429222ee03d..1eeec9513c9907439d9607c18b2ba96b1890465c
aa309dcd59e0f41c597b068ad7ffdf866faa6101..56648c55cb0182ea72a111dc0e4a2e9fa2803eb2

From 7592d0a40715b9c7b6949d888297a7cf59d0afbd Mon Sep 17 00:00:00 2001
From: Robert Walker <bob.mt.wya@gmail.com>
Date: Thu, 2 Mar 2023 11:45:57 +0000
Subject: [PATCH] configure.ac: fix 32-bit target build on 64-bit host

Fix Gentoo specific multilib Portage 32-bit build on 64-bit host
system.

See: https://bugs.gentoo.org/show_bug.cgi?id=395615

Explicitly add the required -m32/m64 to *FLAGS; this overrides any
arch-specific -m* flags that may have been appended by multilib-portage.

Even though -m32/m64 is now added to *FLAGS, -m32/m64 still has to be
explicitly added to CC and CXX due to wine's build system. For example,
winegcc saves the build-time value of CC and uses it at runtime.
---

 configure.ac | 9 +++++++++
 1 file changed, 9 insertions(+)


diff --git a/configure.ac b/configure.ac
index 86f12c598d2..b62d1a3ac06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,4 +132,7 @@ case $host in
       CC="$CC -m64"
       CXX="$CXX -m64"
+      CFLAGS="$CFLAGS -m64"
+      LDFLAGS="$LDFLAGS -m64"
+      CXXFLAGS="$CXXFLAGS -m64"
       host_cpu="x86_64"
       notice_platform="64-bit "
@@ -138,4 +141,7 @@ case $host in
       CC="$CC -m32"
       CXX="$CXX -m32"
+      CFLAGS="$CFLAGS -m32"
+      LDFLAGS="$LDFLAGS -m32"
+      CXXFLAGS="$CXXFLAGS -m32"
       host_cpu="i386"
       notice_platform="32-bit "
@@ -150,4 +156,7 @@ case $host in
       CC="$CC -m32"
       CXX="$CXX -m32"
+      CFLAGS="$CFLAGS -m32"
+      LDFLAGS="$LDFLAGS -m32"
+      CXXFLAGS="$CXXFLAGS -m32"
       AC_MSG_CHECKING([whether $CC works])
       AC_LINK_IFELSE([AC_LANG_PROGRAM()],AC_MSG_RESULT([yes]),
--
